flytekit-java/src/main/java/org/flyte/flytekit/SdkPartialTransform.java

```diff
@@ -16,72 +16,42 @@
  */
 package org.flyte.flytekit;
 
-import static java.util.Collections.emptyList;
-import static java.util.Collections.emptyMap;
 import static java.util.Collections.unmodifiableList;
-import static java.util.Collections.unmodifiableMap;
 import static java.util.Objects.requireNonNull;
 
 import java.time.Duration;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 import java.util.function.Function;
 import javax.annotation.Nullable;
 
 /** {@link SdkTransform} with partially specified inputs. */
-class SdkPartialTransform<T> extends SdkTransform<T> {
-  private final SdkTransform<T> transform;
-  private final Map<String, SdkBindingData<?>> fixedInputs;
+class SdkPartialTransform<InputT, OutputT> extends SdkTransform<InputT, OutputT> {
```

Comment:
```suggestion
class SdkTransformMetadataDecorator<InputT, OutputT> extends SdkTransform<InputT, OutputT> {
```
Reading the java doc made me realize that the name is no longer relevant, given that we no longer hold inputs in the wrapper class. We need to rename the class and change the now obsolete doc